core/field_image.ts

```diff
@@ -248,15 +246,36 @@ export class FieldImage extends Field {
    * @nocollapse
    * @internal
    */
-  static fromJson(options: AnyDuringMigration): FieldImage {
+  static fromJson(options: FromJsonConfig): FieldImage {
+    if (!options.src || !options.width || !options.height) {
```

Comment:
Would it work to just make the fields in the FromJsonConfig non-optional? I realize you'd still want the non-zero check though.

Reply:
We cannot because then the `FieldImage`'s  `fromJson` becomes stricter than the supertype `Field`'s `fromJson`, breaking subtyping rules :/